CONTRIBUTING.md

```diff
@@ -1,256 +1,269 @@
-own.
+`chartpress.yaml` itself has changed.
 
-#### a) Automated Helm chart install or upgrade
+1. Ensure the latest git tags are available locally, as `chartpress` use them.
 
-1. Install or upgrade your local Helm chart.
-   
    ```shell
-   ./dev upgrade
+   git fetch origin --tags
    ```
 
-#### b) Manual Helm chart install or upgrade
-
 1. Use [`chartpress`](https://github.com/jupyterhub/chartpress) to rebuild
-   modified images if needed but also update the chart's
+   images that needs to be rebuilt and update the chart's
    [values.yaml](jupyterhub/values.yaml) file with the appropriate image tags.
 
    ```shell
+   # run this if you are using k3s, or generally when your Kubernetes cluster
+   # can make direct use of your locally built images
    chartpress
-   ```
 
-   > **NOTE:** If you use a kind cluster and have built new images that will
-   > only available locally, you must also load them into the kind cluster using
-   > the `kind load docker-image <image:tag>` command.
+   # run this if you are using k3d
+   chartpress --image-prefix=registry.local:5000/ --push
+   ```
 
-1. Use `helm` to install or upgrade your Helm chart.
+1. Use `helm` to upgrade (and install) your Helm chart.
 
    ```shell
-   helm upgrade jh-dev ./jupyterhub --install --namespace jh-dev
+   # note that --cleanup-on-fail is a very good practice to avoid
+   # "<resource name> already exist" in future upgrades following
+   # a failed upgrade.
+   helm upgrade --install jupyterhub ./jupyterhub --cleanup-on-fail --values dev-config.yaml
    ```
 
+## 5: Visit the JupyterHub
 
+Soon you should be able to visit https://local.jovyan.org:30443 where you
+manually need to choose to accept to trust certificates signed by a certificate
+authority you don't already trust.
 
-### 4: Setup network access
-
-In order for you to access jupyterhub and a spawned user server, you need to be
-able to access the Kubernetes service in this Helm chart called proxy-public.
-While pods in the cluster can do this easily, your computer isn't a pod in the
-cluster. What we can do is to dedicate a port on your computer to go towards the
-proxy-public service of the Kubernetes cluster using `kubectl port-forward`.
+## 6: Run tests
 
-When you run `kubectl port-forward` you will get a process that keeps running
-and you need to open an new terminal window alongside it, unless you detach this
-process. The `./dev port-forward` script will detach the process, and respect
-the environment two variables, `Z2JH_PORT_FORWARD_ADDRESS` and
-`Z2JH_PORT_FORWARD_PORT`, that you can set with the `.env` file.
-
-#### a) Using dev script
+The test suite runs outside your Kubernetes cluster.
 
 ```shell
-./dev port-forward
+pytest -vx -m "not netpol" ./tests
 ```
 
-#### b) Using kubectl directly
+Note that we have disabled the creation of NetworkPolicy Kubernetes resources in
+our [dev-config.yaml](dev-config.yaml) and we choose to not run a test of them
```

Comment:
I thought you fixed this by installing calico/canal?

Reply:
Outdated remnant, good catch! Calico btw, i failed after lots of attempt with k3s in docker mode + canal on travis